File: copperfield-core/src/main/java/dev/volix/rewinside/odyssey/common/copperfield/annotation/CopperIgnore.java
package dev.volix.rewinside.odyssey.common.copperfield.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Flags the field to be ignored by the conversion process of CopperConvertableConverters.
 *
 * The {@link #contextTypes()} define all context types for which this field will be ignored. By default, this field will be ignored for all context
 * types.
 *
 * @author Benedikt Wüller
 */
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
public @interface CopperIgnore {
    Class[] contextTypes() default { Object.class };
}

File: copperfield-core/src/main/java/dev/volix/rewinside/odyssey/common/copperfield/annotation/CopperIgnore.kt
package dev.volix.rewinside.odyssey.common.copperfield.annotation

import dev.volix.rewinside.odyssey.common.copperfield.Convertable
import kotlin.reflect.KClass

/**
 * Excludes this field from conversion to and from the given [Convertable] types.
 *
 * @author Benedikt Wüller
 */
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
annotation class CopperIgnore(vararg val convertables: KClass<out Convertable>)
